src/main/java/seedu/foodrem/model/item/ItemBoughtDate.java

```diff
@@ -40,21 +52,9 @@ public ItemBoughtDate(String dateString) {
      */
     @Override
     public boolean equals(Object other) {
-        if (other == this) {
-            return true;
-        }
-
-        if (!(other instanceof ItemBoughtDate)) {
-            return false;
-        }
-
-        ItemBoughtDate date = (ItemBoughtDate) other;
-
-        if (date.boughtDate == null && boughtDate == null) {
-            return true;
-        }
-
-        return boughtDate.equals(((ItemBoughtDate) other).boughtDate);
+        return other == this // short circuit if same object
+                || (other instanceof ItemBoughtDate // instanceof handles nulls
```

Comment:
Although `LocalDate.MIN` is out of range for us, it's good to be defensive and guard against other instances that are not the unset instance.

```suggestion
                || (other != NOT_SET_BOUGHT_DATE
                && other instanceof ItemBoughtDate // instanceof handles nulls
```

Reply:
Was initially thinking if I do that, two items with the bought date unset will not be equal one another as equals method will return false if both. But I realised since they are referencing the same bought date object it will return true. Thanks for letting me know!